sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/RewriteDistinctAggregates.scala

```diff
@@ -118,7 +118,75 @@ import org.apache.spark.sql.types.IntegerType
  *       LocalTableScan [...]
  * }}}
  *
- * The rule does the following things here:
+ * Third example: single distinct aggregate function with filter clauses (in sql):
+ * {{{
+ *   SELECT
+ *     COUNT(DISTINCT cat1) FILTER (WHERE id > 1) as cat1_cnt,
+ *     COUNT(DISTINCT cat2) as cat2_cnt,
+ *     SUM(value) AS total
+ *  FROM
+ *    data
+ *  GROUP BY
+ *    key
+ * }}}
+ *
+ * This translates to the following (pseudo) logical plan:
+ * {{{
+ * Aggregate(
+ *    key = ['key]
+ *    functions = [COUNT(DISTINCT 'cat1) with FILTER('id > 1),
+ *                 COUNT(DISTINCT 'cat2),
+ *                 sum('value)]
+ *    output = ['key, 'cat1_cnt, 'cat2_cnt, 'total])
+ *   LocalTableScan [...]
+ * }}}
+ *
+ * This rule rewrites this logical plan to the following (pseudo) logical plan:
+ * {{{
+ *   Aggregate(
```

Comment:
This plan rewriting LGTM. Shall we update the second example to make it consistent with this example?
```
   ...
   Aggregate(
      key = ...
      functions = [sum('value)]
      output = ...
     Expand(
        projections = [('key, null, null, 0, if ('id > 1) cast('value as bigint) else null, 'id),
                       (...),
                       (...)]
        output = [...])
       LocalTableScan [...]
```

Reply:
This means to revert some code like
https://github.com/apache/spark/commit/d59e7195f6051820c95f84c83b01148975412d85#diff-8075628c8d39481271059f6fb2dd1e62L160
?

Reply:
OK. I know you means.